numpy/core/src/umath/ufunc_object.c

```diff
@@ -827,9 +831,23 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
                 case 'a':
                     /* possible axes argument for generalized ufunc */
                     if (out_axes != NULL && strcmp(str, "axes") == 0) {
+                        if (out_axis != NULL && *out_axis != NULL) {
+                            PyErr_SetString(PyExc_RuntimeError,
```

Comment:
Why is this RuntimeError here but TypeError above (in override.c)?  I would expect both to give the same error (probably TypeError).

Reply:
That was for symmetry with `sig` and `signature`, but I must admit a `TypeError` makes much more sense, so I've changed it.